### PR TITLE
add gitlab ci template

### PR DESCRIPTION
**Pull request by johnnychen94**
1. test the pulled codes against to different Julia version. (I use a very simple [docker image](https://github.com/johnnychen94/docker-juliabook) made by myself)
2. when all tests pass, compile the book
3. after compilation, a downloadable pdf file as an artifact is provided.

CI demo:
image

Please enlighten me if there's a better tool than `install_pkg.jl`.

### CODE REVIEW

**Comment by tawheeler** [comment]
Thanks @johnnychen94! I will check this over the next few days. I am less familiar with gitlab - great to learn new tools.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  - cd lexer &&
+    python3 setup.py install &&
+    cd ..
+  # update pythontex to 0.17-dev
+  - git clone https://github.com/gpoore/pythontex.git &&
+    cd pythontex/pythontex &&
+    echo 1 | python3 pythontex_install.py &&
+    cd ../..
+  # install julia dependency
+  - julia install_pkg.jl REQUIRE
```
Comment (tawheeler) [inline .gitlab-ci.yml]: should this be `install_pkgs.jl`? IE, with an 's'?

**Comment by tawheeler** [comment]
I am not convinced that we need to test against old versions of Julia, but it is not a bad thing to do.

**Comment by johnnychen94** [comment]
My idea here is to make sure the code samples are runnable in all Julia versions and compile the results using the latest stable version. The three tests are running parallelly, so basically there's no overhead introduced. As a template of `gitlab-ci.yml`, it would be not a big issue though 😄 

I wrote the `install_pkgs.jl` several months ago when I don't know Julia much, as you might see it isn't good enough - *it just works*. We might need to find a better tool or rewrite this. The purpose is to install dependency packages to Julia base version.

**Comment by tawheeler** [comment]
This is great, thanks!

**Pull request merged**